framework/util/page_guard_manager.cpp

```diff
@@ -230,15 +235,58 @@ PageGuardManager::~PageGuardManager()
     }
 }
 
+#if !defined(WIN32)
+void* PageGuardManager::signal_handler_watcher_f(void* args)
+{
+    while (instance_->signal_handler_watcher_)
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(PageGuardManager::signal_watcher_thread_period));
+
+        struct sigaction current_handler;
+        int              result = sigaction(SIGSEGV, nullptr, &current_handler);
+
+        assert(instance_);
+        if (result != -1 && instance_->exception_handler_ && current_handler.sa_sigaction != PageGuardExceptionHandler)
+        {
+            assert(instance_->exception_handler_count_);
+
+            GFXRECON_LOG_WARNING("Signal handler has been removed. Re-installing.")
```

Comment:
`GFXRECON_LOG_WARNING_ONCE`?

Reply:
I think it it would be useful to know how often an application install its own handlers

Reply:
True. Especially for us as the maintainers. If it is only at init time this is fine. If some other app does it every frame and the user has to lower the sleep period to catch it, this logging will slow gfxr down. I'm not sure if the user would then have to rebuild gfxr to kill the perf cost of the logging or whether a `grep -v` on the output would be enough.Anyway, this is just a comment.